main.c

```diff
@@ -40,10 +40,11 @@ void gen_trytes(uint16_t len, char *out) {
   }
 }
 
-void send_https_msg(char const *const host, char const *const port, char const *const api, char const *const tryte_msg,
-                    char const *const addr) {
+int send_https_msg(char const *const host, char const *const port, char const *const api, char const *const tryte_msg,
```

Comment:
Can this return type be changed to `retcode_t`?

Reply:
Inifinate blocking seem not a big problem for these commits, I would revert it.